src/System.Text.Json/src/System/Text/Json/Serialization/ReadStackFrame.cs

```diff
@@ -54,13 +57,24 @@ internal void ResetProperty()
             PopStackOnEndArray = false;
             EnumerableCreated = false;
             TempEnumerableValues = null;
+            KeyName = null;
+        }
+
+        internal bool IsProcessingEnumerableOrDictionary()
+        {
+            return IsEnumerable() ||IsPropertyEnumerable() || IsDictionary() || IsPropertyADictionary();
```

Comment:
This bool helper ends up repeating certain unnecessary null checks (i.e. `JsonPropertyInfo != null`)